lib/phoenix_live_view/html_tokenizer.ex

```diff
@@ -94,11 +94,12 @@ defmodule Phoenix.LiveView.HTMLTokenizer do
   end
 
   defp handle_text("<!--" <> rest, line, column, buffer, acc, state) do
+    state = Map.put(state, :context, :comment_start)
```

Comment:
@josevalim I'm not 100% sure if that is the correct approach.  I will wait for your feedback so I can do the same for the `script` tag 🙌🏻  and finish this PR.

Reply:
I added in the state to get it working faster but I guess we shouldn't do it otherwise we might have the wrong state in case we don't clean up. Maybe a better way would be changing the function signature to something like:

```elixir
  defp handle_text("<!--" <> rest, meta, buffer, acc, state) do
    meta = %{line: line, column: column, context: :comment}
    handle_comment(rest, meta, ["<!--" | buffer], acc, state)
  end
```

But that would require more changes since we would need to change all functions.

Reply:
I think you can keep it in the state, as it is rarely updated. However, you need to keep all starting and ending tokens, because in the same text you can have HTML comment starting and finishing. And then you prune it down at the end. For example, when the comment starts, you would do:

```elixir
put_in state.context, &[:comment_start | &1]
```

And when it finishes:

```
put_in state.context, &[:comment_end | &1]
```

And then at the end, you want to pop it from the state and remove closed contexts:

```elixir
defp reverse_context([:comment_end, :comment_start | rest], acc), do: reverse_context(rest, acc)
defp reverse_context([:script_end, :script_start | rest], acc), do: reverse_context(rest, acc)
defp reverse_context([head | rest], acc), do: reverse_context(rest, [head | acc])
defp reverse_context([], acc), do: acc
```

And then you add the context to meta if it is not an empty list. The complicated tests are stuff like this:

1. Comment starts and end in the same text

```html
Some <!-- comment --!> text
```

2. Comment ends and script starts in the same text

```html
<!-- <%= "Hello" %> --><script><%= "World" %></script>
```